src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -159,28 +169,33 @@ internal bool TryAddWithoutValidation(HeaderDescriptor descriptor, IEnumerable<s
                 throw new ArgumentNullException(nameof(values));
             }
 
-            using (IEnumerator<string?> enumerator = values.GetEnumerator())
+            using IEnumerator<string?> enumerator = values.GetEnumerator();
+            if (enumerator.MoveNext())
             {
+                TryAddWithoutValidation(descriptor, enumerator.Current);
                 if (enumerator.MoveNext())
                 {
-                    TryAddWithoutValidation(descriptor, enumerator.Current);
-                    if (enumerator.MoveNext())
+                    ref object? storeValueRef = ref GetValueRefOrAddDefault(descriptor);
```

Comment:
It seems like you could optimize this further by doing the GetValueRefOrAddDefault on the first value instead of the second. As it stands, if there is more than one value we will end up doing two lookups -- one for the TryAddWithoutValidation above, and another for the GetValueRefOrAddDefault here.

Reply:
It could, with duplicating some of the code in `TryAddWithoutValidation(HeaderDescriptor, string)`.
I wasn't sure how much we care about the perf of this method (e.g. we don't try to avoid boxing the enumerator either).

Reply:
Honestly I doubt we care that much.